### ISSUE TITLE
screen-inhibit@mtwebster: No longer prevents screensaver when the monitor was turned off since Mint 21

### ISSUE THREAD

**Issue by IzzySoft** [linuxmint/cinnamon-spices-applets#6079]
### Applet version/Build date

2.0.1

### Cinnamon version

6.0.4

### Distribution

Mint 21.3

### Graphics hardware *and* driver used

AMD Vega (Ryzen 7 5700G with built-in graphics)

### Applet name and maintainer

screen-inhibit@mtwebster @claudiux 

### What happened?

The applet worked great in previous Mint releases (up to 20.3). But since I've updated to Mint 21 (and now 21.3), whenever the monitor (hardware) was switched off, I need to restart the applet (right-click its icon in the tray, "Configure", hamburger menu, "reload screen-inhibit") to have it preventing the screensaver from kicking in. The screen is however still prevented from being locked it seems.

After I reload the applet, the screensaver itself won't kick in anymore either. If it matters: "Inhibit screensaver at startup" is toggled on.

As outlined, my machine uses a Ryzen 7 5700G with built-in graphics and the default driver Mint installs for it (so the FOSS one, not any manually installed proprietary driver).

**Comment by rcalixte** [comment]
@IzzySoft What applet are you referring to?

**Comment by IzzySoft** [comment]
Oof, I've opened the issue via the applet page, so I thought the template applied that accordingly, apologies! `screen-inhibit@mtwebster` it is.

**Comment by rcalixte** [comment]
> PS: Funny. I just used the "new issue" button above to check, and now there's a mandatory field to pick the applet name and maintainer from. I didn't see that when I filed this issue (as the field is mandatory, I couldn't have saved the issue without otherwise).

The field was mandatory before but it was not defaulting to a null value which appears to have been confusing for some. Hopefully the empty default is better going forward.

**Comment by rcalixte** [comment]
cc @claudiux

**Comment by IzzySoft** [comment]
> defaulting to a null value

eek, then that was how I missed it. Apologies again, should have checked once more before hitting submit!

**Comment by rcalixte** [comment]
> eek, then that was how I missed it. Apologies again, should have checked once more before hitting submit!

You weren't alone! Hence the need for the change. The good news is that it is apparent when you go to the form now. Hopefully this means it will be clearer longer-term for others.

**Comment by claudiux** [comment]
I'm working on that. Please be patient.

**Comment by IzzySoft** [comment]
Will be, thanks a lot! Didn't know you're already on it…

**Comment by claudiux** [comment]
@IzzySoft 
Please try #6099 by copying `applet.js` and `settings-schema.js` into your `~/.local/share/cinnamon/applets/screen-inhibit@mtwebster` directory. Restart Cinnamon to test.
I'm waiting for your feedback before publishing this version.

**Comment by IzzySoft** [comment]
Replaced `applet.js` and `settings-schema.json` from the PR (there was no `settings-schema.js` so I assume you've meant the JSON) and restarted Cinnamon. Now waiting what happens. Will update here later, just wanted to let you know I'm on it.

**Comment by IzzySoft** [comment]
OK, short update: This seems to work as hoped now, thanks! Monitor was off over night, and after turning it on again (without restarting the service) I've left the machine "untouched" for 2h during a conference. Screen saver was successfully inhibited again.

I see there already is an update (arrived yesterday shortly after your above comment). Did you already push it, or should I rather skip that update and wait for the next one?

**Comment by claudiux** [comment]
The update was for a translation (not mine). When I put my version online, this issue will be automatically closed.

**Comment by IzzySoft** [comment]
Thanks @claudiux! I see you've merged, so I guess I wait a little to be on the safe side before accepting the "official update"? I've no idea how long it takes from "merge" to "being offered for download", but am totally fine to wait a few more days (especially as I now have a working version in place, thanks to you).

**Comment by IzzySoft** [comment]
It currently offers a 2024.06.19, which means "today". Guess that is the one already then?

**Comment by claudiux** [comment]
Once updated, you should see in `About...` (context menu of this applet): `Version 2.1.0 (2024-06-19 14:18:47)`

**Comment by IzzySoft** [comment]
Now that's strange. Yesterday it worked as described (with the 2 files from the PR). I changed nothing since (so I didn't yet apply the offered update). Today the screensaver kicks in again despite of inhibitor showing it's active :disappointed:

**Comment by claudiux** [comment]
Is your computer a laptop or a desktop?

**Comment by IzzySoft** [comment]
> Is your computer a laptop or a desktop?

Desktop, on permanent power and running 24/7. I just turn off the monitor for the night (at night, some batch jobs run). Note sure if something changed making it to interpret "monitor off" as "lid close" – but the screen is not locked, just the screensaver kicks in.

> Please show here the output of this command:

<details><summary><code>gsettings list-recursively | grep org.cinnamon.settings-daemon.plugins.power</code></summary><pre>
org.cinnamon.settings-daemon.plugins.power backlight-helper-force false
org.cinnamon.settings-daemon.plugins.power backlight-helper-preference-order ['firmware', 'platform', 'raw']
org.cinnamon.settings-daemon.plugins.power button-hibernate 'hibernate'
org.cinnamon.settings-daemon.plugins.power button-power 'interactive'
org.cinnamon.settings-daemon.plugins.power button-suspend 'suspend'
org.cinnamon.settings-daemon.plugins.power critical-battery-action 'hibernate'
org.cinnamon.settings-daemon.plugins.power idle-brightness 30
org.cinnamon.settings-daemon.plugins.power idle-dim-ac false
org.cinnamon.settings-daemon.plugins.power idle-dim-battery true
org.cinnamon.settings-daemon.plugins.power idle-dim-time 90
org.cinnamon.settings-daemon.plugins.power inhibit-lid-switch true
org.cinnamon.settings-daemon.plugins.power lid-close-ac-action 'suspend'
org.cinnamon.settings-daemon.plugins.power lid-close-battery-action 'suspend'
org.cinnamon.settings-daemon.plugins.power lid-close-suspend-with-external-monitor false
org.cinnamon.settings-daemon.plugins.power lock-on-suspend true
org.cinnamon.settings-daemon.plugins.power minimum-display-brightness uint32 2
org.cinnamon.settings-daemon.plugins.power percentage-action 3
org.cinnamon.settings-daemon.plugins.power percentage-critical 4
org.cinnamon.settings-daemon.plugins.power percentage-low 10
org.cinnamon.settings-daemon.plugins.power power-notifications-for-keyboard true
org.cinnamon.settings-daemon.plugins.power power-notifications-for-mouse true
org.cinnamon.settings-daemon.plugins.power power-notifications-for-other-devices true
org.cinnamon.settings-daemon.plugins.power sleep-display-ac 0
org.cinnamon.settings-daemon.plugins.power sleep-display-battery 1800
org.cinnamon.settings-daemon.plugins.power sleep-inactive-ac-timeout 0
org.cinnamon.settings-daemon.plugins.power sleep-inactive-ac-type 'suspend'
org.cinnamon.settings-daemon.plugins.power sleep-inactive-battery-timeout 0
org.cinnamon.settings-daemon.plugins.power sleep-inactive-battery-type 'suspend'
org.cinnamon.settings-daemon.plugins.power time-action 240
org.cinnamon.settings-daemon.plugins.power time-critical 420
org.cinnamon.settings-daemon.plugins.power time-low 1200
org.cinnamon.settings-daemon.plugins.power use-time-for-policy false
</pre></details>

**Comment by claudiux** [comment]
@IzzySoft 
The only notable difference with mine: `lock-on-suspend false`. But it doesn't seem very important.

**Comment by claudiux** [comment]
Ok, I found a solution.

**Comment by claudiux** [comment]
Please try v2.1.1.

**Comment by IzzySoft** [comment]
> `lock-on-suspend`

The funny thing is the screen does not get locked, only "screensaved" (or maybe I didn't wait long enough).

> Ok, I found a solution.

Yay! :partying_face: 

> Please try v2.1.1.

Will do! I see you've already pushed it – so I'll update as soon as it shows up here. Will report back then after 2..3 days of testing. Thanks a lot again!

**Comment by IzzySoft** [comment]
Nope, sorry: screensaver kicked in again and, as this time I was "AFK" a bit longer, screen was locked.

Not sure if it's related, but I think cannot hurt to mention: while the screen is turned off, the notification applet (in the tray) also seems to stop collection notifications. Turning the screen back on shows it empty despite of "new events" waiting in different applications. As pointed out, all this started only after upgrading from Mint 20.3, was working fine back then.

**Comment by claudiux** [comment]
Ok, thanks for the feedback. I'll try another solution.

**Comment by claudiux** [comment]
v2.1.2 introduces two aggressive ways of blocking the screensaver. In this applet's settings, select Aggressive mode and set the interval (in minutes) between two screensaver blockings.
If Aggressive mode doesn't work, try Super Aggressive mode.

**Comment by IzzySoft** [comment]
Thanks, updated and set "Aggressive" with an interval slightly below the time the screensaver would kick in normally. Will watch this a while and provide feedback as usual.

I must say: great support, thanks a lot! Not something I'd take for granted. I wonder if I'm the only one with this issue…

**Comment by claudiux** [comment]
Thanks for helping me to improve this applet.

**Comment by IzzySoft** [comment]
Unfortunately neither mode helped. I first tried "Aggressive" with 10 min interval (screensaver is set to kick in after 15 min), then "Super Aggressive". Still, once the monitor was turned off, the old picture when having it turned on again: screensaver kicks in after 15 min.

I meanwhile rather think it's some bug with Cinnamon and not your Inhibitor. As I wrote, notifications aren't collected either once the monitor was turned of. As if Cinnamon then somehow "silences" applets and then forgets to wake them up again (wish I had an idea for good search terms to figure this – hard to believe it should only be me affected here, but I've not yet heard/read about it anywhere).

So for now I've decided to simply disable the screensaver entirely. I can still lock the screen manually when needed.

**Comment by rcalixte** [comment]
> Unfortunately neither mode helped.

Have you tried the built-in inhibit@cinnamon applet? I'm curious if this will work (and identify a bug in this applet) or if it won't work (and identify a potential system issue).

**Comment by claudiux** [comment]
Please try "Aggressive" with 15 min interval, then with 1 min interval.

What is the return of the command: `ls -l $(which cinnamon-screensaver-command)`?

**Comment by IzzySoft** [comment]
Now it gets funny: in screensaver settings, I've set the delay to "Never", meaning it should not turn itself on at all. You'd think. But no, when I was AFK for ~15 minutes and came back, screensaver was active.

![image](https://github.com/linuxmint/cinnamon-spices-applets/assets/6781438/7defc915-073c-4d8f-9e6d-db8b828d9491)

Despite of that idle-activation is enabled, as dconf shows:

![image](https://github.com/linuxmint/cinnamon-spices-applets/assets/6781438/59cb3dc9-42d5-4c6a-a109-995b1efe0e20)
![image](https://github.com/linuxmint/cinnamon-spices-applets/assets/6781438/d35214f2-13a8-489b-bf0b-c7349c405834)

Funny I can find no field for the idle-timeout there. I'll leave it at that for now to test the settings you've just suggested. As for the command:

`-rwxr-xr-x 1 root root 215 Jan 24 16:25 /usr/bin/cinnamon-screensaver-command` – so the very same here.

I've now set "Super Aggressive" with 15 min. Will try 1 min should that fail. And of course report back if it seems to work.

**Comment by IzzySoft** [comment]
> Have you tried the built-in inhibit@cinnamon applet?

@rcalixte sorry, I've missed your comment. No, I have not. But if the screensaver even kicks in when I set it not to, I have my doubts. Can try it after the current tests are through. But looking at it's description, a question to @claudiux:

> This applet is used to disable notifications and power management

"disable notifications" is what happens here unwantedly (it didn't on 20.3). Does yours disable notifications as well? On 20.3 they were shown even on the screensaver, if I remember correctly.

**Comment by claudiux** [comment]
`screen-inhibit@mtwebster` does not disable notifications. `inhibit@cinnamon.org` can disable screensaver, or notifications or both.

**Comment by IzzySoft** [comment]
I will see tomorrow morning how super-aggressive with 15min works. Then can chack on Friday morning with 1min. After that, I'm AFK for about 10 days, so please do not be irritated about a gap in communication from my side then.
